### DISCUSSION TITLE
21634 tgfx_JNIEnvironment

### DISCUSSION THREAD

**Discussion by lsbnft0** [Tencent/libpag#2520] [General]
系统版本： Android 10,level 29   
ROM： HuaWei/EMOTION
CPU架构： arm64-v8a
设备机型： HONOR_YAL-AL10
集成方式：在线集成，版本号：4.3.57

#00 pc 000000000007066c /apex/com.android.runtime/lib64/bionic/libc.so (abort+160) [arm64-v8a::b91c775ccc9b0556e91bc575a2511cd0]

#01 pc 00000000000d0d4c /apex/com.android.runtime/lib64/bionic/libc.so (__fortify_fatal(char const*, ...)+116) [arm64-v8a::b91c775ccc9b0556e91bc575a2511cd0]

#02 pc 00000000000d0480 /apex/com.android.runtime/lib64/bionic/libc.so (HandleUsingDestroyedMutex(pthread_mutex_t*, char const*)+52) [arm64-v8a::b91c775ccc9b0556e91bc575a2511cd0]

#03 pc 00000000000d02e4 /apex/com.android.runtime/lib64/bionic/libc.so (pthread_mutex_lock+148) [arm64-v8a::b91c775ccc9b0556e91bc575a2511cd0]

#04 pc 000000000023029c /data/app/com.notedance.nn-c6-i8B_osTv05xpJjZ-QGA==/lib/arm64/libpag.so [arm64-v8a::60e8e56c058f4ef6ea96caec7b160997]

#05 pc 00000000000e145c /data/app/com.notedance.nn-c6-i8B_osTv05xpJjZ-QGA==/lib/arm64/libpag.so (pag::PAGLayer::duration() const+68) [arm64-v8a::60e8e56c058f4ef6ea96caec7b160997]

#06 pc 000000000010aedc /data/app/com.notedance.nn-c6-i8B_osTv05xpJjZ-QGA==/lib/arm64/libpag.so (Java_org_libpag_PAGLayer_duration+68) [arm64-v8a::60e8e56c058f4ef6ea96caec7b160997]

#07 pc 000000009e50d77c <unknown>

java:
org.libpag.PAGImageView.onAnimationUpdate(SourceFile:5)

org.libpag.PAGAnimator.onAnimationUpdate(SourceFile:3)

**Comment by lsbnft0** [comment]
手机系统层抛出异常，让app闪退了。少部分的情况才出现。

**Comment by lsbnft0** [comment]
@domchen 大佬请帮分析下哈

**Reply by domchen** [reply]
报错的提示是使用了析构掉的线程锁，所以应该是整个PAGLayer实例都是个野指针。检查一下上层业务逻辑是否存在生命周期管理的问题，访问了销毁的实例对象。
